### PR TITLE
fix(text-editor): ensure the input is focused when toolbar buttons are activated or deactivated

### PR DESCRIPTION
**Pull request by edleeks87**
### Proposed behaviour

Ensures that the input is refocused when a user interacts with the toolbar buttons to add remove styling to the rich text.

### Current behaviour

Input is only refocused when activating a styling button, focus remains on the deactivated button otherwise and when keyboard navigation is used it focuses buttons out of sequence.

### Checklist

- [x] Commits follow our style guide
- [x] Playwright automation tests added or updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Additional context

There is a bug in `draft-js` whereby some callbacks are called on focus whist others are not called when expected. In this case onBlur is not being called as expected meaning the focus flag held in state does not update but the index used to handle focusing toolbar buttons is reset. I have chosen to align the behaviour so that the input is focused when a toolbar button is interacted with by a user so it is consistent and also avoids the above issue.

### CODE REVIEW

**Review by nineteen88 — DISMISSED** [review]

File: src/components/text-editor/text-editor.pw.tsx [hunk 1]
```diff
@@ -199,6 +180,74 @@ test.describe("Functionality tests", () => {
     });
   });
 
+  buttonNames.forEach((buttonType, times) => {
+    test(`should focus the input when ${buttonType} is selected/ deselected with enter key`, async ({
```
Comment (DipperTheDan) [inline src/components/text-editor/text-editor.pw.tsx:184]: **nitpick:** Most nitpicky thing ever. Trailing space after `/`

```suggestion
    test(`should focus the input when ${buttonType} is selected/deselected with enter key`, async ({
```

File: src/components/text-editor/text-editor.pw.tsx [hunk 2]
```diff
@@ -199,6 +180,74 @@ test.describe("Functionality tests", () => {
...
+        await page.keyboard.press("ArrowRight");
+      }
+      await expect(textEditorToolbar(page, buttonType)).toBeFocused();
+      await page.keyboard.press("Enter");
+      await expect(textInput).toBeFocused();
+    });
+  });
+
+  buttonNames.forEach((buttonType, times) => {
+    test(`should focus the input when ${buttonType} is selected/ deselected with space key`, async ({
```
Comment (DipperTheDan) [inline src/components/text-editor/text-editor.pw.tsx:210]:
```suggestion
    test(`should focus the input when ${buttonType} is selected/deselected with space key`, async ({
```

File: src/components/text-editor/text-editor.pw.tsx [hunk 3]
```diff
@@ -199,6 +180,74 @@ test.describe("Functionality tests", () => {
...
+        await page.keyboard.press("ArrowRight");
+      }
+      await expect(textEditorToolbar(page, buttonType)).toBeFocused();
+      await page.keyboard.press("Space");
+      await expect(textInput).toBeFocused();
+    });
+  });
+
+  buttonNames.forEach((buttonType) => {
+    test(`should focus the input when ${buttonType} is selected/ deselected via user clicking`, async ({
```
Comment (DipperTheDan) [inline src/components/text-editor/text-editor.pw.tsx:236]:
```suggestion
    test(`should focus the input when ${buttonType} is selected/deselected via user clicking`, async ({
```

**Review by nineteen88 — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Pull request merged**